### PR TITLE
Postcode validation

### PR DESCRIPTION
**Pull request by Jaber-aldroubi**

### CODE REVIEW

File: app/src/main/java/com/example/amcc/view/MainFragment.java [hunk 1]
```diff
@@ -201,10 +203,10 @@ private boolean inputIsValid() {
 
         validation.addValidation(avgConField, "(^[1-9]\\d?(\\.[0-9]\\d?)?$)", errorMsg);
         View error = city;
-        if (!cityFromApi.contains(city.getText().toString())) {
-
+        String location = city.getText().toString();
+        if (!cityFromApi.contains(location) && !viewModel.mPostCodes.contains(location)) {
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/view/MainFragment.java:206]: Why is the if statment && und not || ?

File: app/src/main/java/com/example/amcc/viewModel/SharedViewModel.java [hunk 2]
```diff
@@ -39,11 +41,26 @@ public void onResponse(Call<List<String>> call, Response<List<String>> response)
...
+            @Override
+            public void onResponse(Call<List<String>> call, Response<List<String>> response) {
+                if (response.isSuccessful() && response.body() != null) {
+                    mPostCodes.addAll(response.body());
+                }
+            }
+
+            @Override
+            public void onFailure(Call<List<String>> call, Throwable t) {
+            }
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/viewModel/SharedViewModel.java:60]: Need to set value of cityError here?

File: app/src/main/java/com/example/amcc/view/MainFragment.java (hunk 1 above)
Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/view/MainFragment.java:206]: if it is not a city and it is not a postcode then show error

File: app/src/main/java/com/example/amcc/viewModel/SharedViewModel.java (hunk 2 above)
Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/viewModel/SharedViewModel.java:60]: nope

**Review by Shaftar — APPROVED** [review]

**Pull request merged**